src/dynamics/initial_conditions.jl

```diff
@@ -313,20 +317,31 @@ function initialize!(
     initial_conditions::RossbyHaurwitzWave,
     model::AbstractModel,
 )
-    (; m, ω, K, c) = initial_conditions
+    (; m, ω, K, Ω, a, g, h0, c) = initial_conditions
     (; geometry) = model
 
     # Rossby-Haurwitz wave defined through vorticity ζ as a function of
     # longitude λ, latitude θ (in degrees), sigma level σ (vertically constant though)
     # see Williamson et al. 1992, J Computational Physics, eq 145
     ζ(λ, θ, σ) = 2ω*sind(θ) - K*sind(θ)*cosd(θ)^m*(m^2 + 3m + 2)*cosd(m*λ)
+    # see Williamson et al. 1992, J Computational Physics, eq 147 - 149, 146
+    A(λ, θ) = ω/2 * (2Ω+ω)*cosd(θ)^2 + K^2/4*cosd(θ)^(2m)*((m+1)*cosd(θ)^2 + (2m^2-m-2) - 2m^2/(cosd(θ)^2))
+    B(λ, θ) = (2(Ω+ω)*K)/((m+1)*(m+2))*cosd(θ)^m*( (m^2+2m+2) - (m+1)^2*cosd(θ)^2 )
+    C(λ, θ) = K^2/4*cosd(θ)^(2m)*((m+1) * cosd(θ)^2 - (m + 2))
+    #η(λ, θ) = h0 + a^2/g*(A(λ,θ) + B(λ,θ)*cosd(m*λ) + C(λ,θ)*cosd(2m*λ))
+    η(λ, θ) = a^2/g*(A(λ,θ) + B(λ,θ)*cosd(m*λ) + C(λ,θ)*cosd(2m*λ))
+
     set!(progn, geometry, vor = ζ)
+    set!(progn, geometry, pres = η)
```

Comment:
alternatively (and I think that's the easier solution) you can put a `model isa ShallowWater &&` in front of this line, so only in the case of the model being a shallow water model the `pres` is actually set -- otherwise it just remains zero.

Reply:
But then went for the conditional `set!` of eta here, I like!